### PR TITLE
cleaner variables incl variable template

### PR DESCRIPTION
**Pull request by dariuszparys**
* Refactoring Variable Code Base Usage
    Simplified and cleaned-up the usage of variables inside all python files
and Azure DevOps pipelines. Variables are now defined in two places, the
Azure DevOps variable template and the Azure DevOps variable group.
* Refactored dotenv usage in a singleton class to serve variables to all
python scripts
* Created new variable template `azdo-variables.yml` to be used within
Azure DevOps pipelines
* Adjusted existing pipelines to leverage new variable template
* Prepared the ARM template to specify individual names for the various
resources
* Adding Workspace in Variable Group
    The workspace has now to be defined in the variable group
* Adjusted infrastructure as code template
* Default WORKSPACE_NAME definition
    In case there is no WORKSPACE_NAME in the Azure DevOps variable group we
build up the workspace using the provided BASE_NAME variable.
* Extended Getting Started Documentation
    Describes the usage of all variable definitions
* Removed unnecessary variables from `azdo-variables.yml`
* `workspace` is now a required ARM template parameter. Needs to exist in the variable group as `WORKSPACE_NAME`
* Cleaned-up `.env.example` to only include variables needed to execute
scripts in local environments
* Modified Getting Started Document
    Included pointers where to find the proper variable options in order to
execute the project locally vs. Azure DevOps

### CODE REVIEW

**Comment by dariuszparys** [comment]
Are forked PRs are able to run the pipeline due to SP_APP_SECRET?

File: .pipelines/azdo-variables.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  value: ''
+- name: REGISTER_SCRIPT_PATH
+  value: register/register_model.py
+- name: SOURCES_DIR_TRAIN
+  value: code
+- name: TRAIN_SCRIPT_PATH
+  value: training/train.py
+- name: TRAINING_PIPELINE_NAME
+  value: training-pipeline
+- name: BUILD_BUILDID
```
Comment (dtzar) [inline .pipelines/azdo-variables.yml]: This and RELEASE_RELEASEID should be removed

File: environment_setup/arm-templates/cloud-environment.json [hunk 2]
```diff
@@ -26,16 +26,36 @@
...
+      "defaultValue": "[concat(toLower(parameters('baseName')),'amlcr')]"
     }
   },
   "variables": {
-    "amlWorkspaceName": "[concat(parameters('baseName'),'-AML-WS')]",
-    "storageAccountName": "[concat(toLower(parameters('baseName')), 'amlsa')]",
+    "amlWorkspaceName": "[parameters('workspace')]",
+    "storageAccountName": "[parameters('storageAccount')]",
     "storageAccountType": "Standard_LRS",
-    "keyVaultName": "[concat(parameters('baseName'),'-AML-KV')]",
```
Comment (dtzar) [inline environment_setup/arm-templates/cloud-environment.json]: Let's keep the `baseName` with suffix for all except workspace

File: environment_setup/iac-create-environment.yml [hunk 3]
```diff
@@ -25,11 +25,11 @@ steps:
   inputs:
     azureSubscription: 'AzureResourceConnection'
     action: 'Create Or Update Resource Group'
-    resourceGroupName: '$(BASE_NAME)-AML-RG'
+    resourceGroupName: '$(RESOURCEGROUP_NAME)'
```
Comment (dtzar) [inline environment_setup/iac-create-environment.yml]: should be `RESOURCE_GROUP`

File: environment_setup/iac-remove-environment.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ steps:
   inputs:
     azureSubscription: 'AzureResourceConnection'
     action: 'DeleteRG'
-    resourceGroupName: '$(BASE_NAME)-AML-RG'
+    resourceGroupName: '$(RESOURCEGROUP_NAME)'
```
Comment (dtzar) [inline environment_setup/iac-remove-environment.yml]: `RESOURCE_GROUP`

**Pull request merged**